arches/app/views/resource.py

```diff
@@ -537,80 +518,55 @@ def get_instance_permissions(self, resource_instance):
         result["creatorid"] = instance_creator["creatorid"]
         return result
 
-    def make_instance_private(self, resourceinstanceid, graphid=None):
-        resource = Resource(resourceinstanceid)
-        resource_instance = models.ResourceInstance.objects.get(pk=resourceinstanceid)
-        resource.graph_id = graphid if graphid else str(resource_instance.graph_id)
-        resource.createdtime = resource_instance.createdtime
-        resource.add_permission_to_all("no_access_to_resourceinstance")
-        instance_creator = resource.get_instance_creator_and_edit_permissions()
-        user = User.objects.get(pk=instance_creator["creatorid"])
-        assign_perm("view_resourceinstance", user, resource)
-        assign_perm("change_resourceinstance", user, resource)
-        assign_perm("delete_resourceinstance", user, resource)
-        remove_perm("no_access_to_resourceinstance", user, resource)
-        return self.get_instance_permissions(resource)
-
-    def make_instance_public(self, resourceinstanceid, graphid=None):
-        resource = Resource(resourceinstanceid)
-        resource_instance = models.ResourceInstance.objects.get(pk=resourceinstanceid)
-        resource.graph_id = graphid if graphid else str(resource_instance.graph_id)
-        resource.createdtime = resource_instance.createdtime
-        resource.remove_resource_instance_permissions()
-        return self.get_instance_permissions(resource)
-
     def apply_permissions(self, data, user, revert=False):
-        with transaction.atomic():
-            for instance in data["selectedInstances"]:
-                resource_instance = models.ResourceInstance.objects.get(
-                    pk=instance["resourceinstanceid"]
-                )
-                for identity in data["selectedIdentities"]:
-                    if identity["type"] == "group":
-                        identityModel = Group.objects.get(pk=identity["id"])
-                    else:
-                        identityModel = User.objects.get(pk=identity["id"])
+        for instance in data["selectedInstances"]:
+            resource_instance = models.ResourceInstance.objects.get(
+                pk=instance["resourceinstanceid"]
+            )
+            for identity in data["selectedIdentities"]:
+                if identity["type"] == "group":
+                    identityModel = Group.objects.get(pk=identity["id"])
+                else:
+                    identityModel = User.objects.get(pk=identity["id"])
 
-                    instance_creator = (
-                        resource_instance.get_instance_creator_and_edit_permissions(
-                            user
+                instance_creator = (
+                    resource_instance.get_instance_creator_and_edit_permissions(user)
```

Comment:
Maybe out of scope for this PR, but it's worth considering that we using the principle user rather than the creator. That would spare a trip to the db because we already have the resource instance.